provider/webhook/webhook.go

```diff
@@ -224,7 +229,11 @@ func (p WebhookProvider) ApplyChanges(ctx context.Context, changes *plan.Changes
 	if resp.StatusCode != http.StatusNoContent {
 		applyChangesErrorsGauge.Inc()
 		log.Debugf("Failed to apply changes with code %d", resp.StatusCode)
-		return fmt.Errorf("failed to apply changes with code %d", resp.StatusCode)
+		err := fmt.Errorf("failed to apply changes with code %d", resp.StatusCode)
+		if resp.StatusCode >= http.StatusInternalServerError {
```

Comment:
```suggestion
		if resp.StatusCode >= http.StatusInternalServerError  && resp.StatusCode <= http.StatusNotExtended {
```

Reply:
I like the suggestion. IMO, the check should be consistent across all three interface functions, therefore I extracted it into a small function for convenience. WDYT?

Reply:
Sounds good to me. Thanks.